.github/workflows/reusable-mypy.yml

```
@@ -0,0 +1,21 @@
+on:
+  workflow_call:
+
+jobs:
+  check-with-mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+
+      - name: Install dependencies and type information
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install mypy
+
+      - name: mypy static type check
+        run: mypy --install-types --ignore-missing-imports --non-interactive --pretty .
```

Without this `mkdir` command I was getting `error: --install-types failed (no mypy cache directory)` when running the `mypy` command locally. Also see https://github.com/python/mypy/issues/10863 and https://github.com/python/mypy/issues/10768 and the [mypy docs](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-cache-dir).
```suggestion
        run: mkdir .mypy_cache && mypy --install-types --ignore-missing-imports --non-interactive --pretty .
```